### PR TITLE
experimentation: return status of an experiment

### PR DESCRIPTION
**Pull request by Augustyniak**
### Description

The PR end up a big bigger than I would like it to be but hopefully the fact that most of the changes in here are pretty mechanical - simple operations - it should be reviewable.

Changes log:

1. Make `CreateOrGetExperiment` and `CreateExperiment` endpoints return the status of an experiment.
2. Remove `STOPPED` status of an experiment to make it easier for users of `Status` enum to work with the type - `STOPPED` introduced a granularity that was not really needed, it is combined with `CANCELED` now.
3. Rename `id` property of ExperimentConfig to `Id`.
4. Make storer return instances of `experimentstore.Experiment` as opposed to `experimentationv1.Experiment`.

### Testing Performed
Unit Tests

### CODE REVIEW

File: backend/module/chaos/experimentation/api/experimentation.go [hunk 1]
```diff
@@ -123,13 +133,21 @@ func (s *Service) CancelExperimentRun(ctx context.Context, req *experimentation.
...
-		s.logger.Errorw("GetExperiments: Unable to retrieve experiments", "error", err)
 		return &experimentation.GetExperimentsResponse{}, err
 	}
 
-	return &experimentation.GetExperimentsResponse{Experiments: experiments}, nil
+	ee := make([]*experimentation.Experiment, len(es))
+	for i := range ee {
+		ep, err := es[i].Proto(time.Now())
+		if err != nil {
+			return nil, err
```
Comment (snowp) [inline backend/module/chaos/experimentation/api/experimentation.go:145]: feels more idiomatic to do a `for i, e := range es` loop

File: backend/service/chaos/experimentation/experimentstore/experiment.go [hunk 2]
```diff
@@ -13,7 +15,7 @@ type Experiment struct {
 	Config *ExperimentConfig
 }
 
-func (rc *Experiment) toProto() (*experimentationv1.Experiment, error) {
+func (rc Experiment) Proto(now time.Time) (*experimentationv1.Experiment, error) {
```
Comment (snowp) [inline backend/service/chaos/experimentation/experimentstore/experiment.go:18]: Feels a bit weird to have a conversion to proto take a current time? Maybe rename this to make it clear that it evaluates the state based on current time?

File: backend/service/chaos/experimentation/experimentstore/storer.go [hunk 3]
```diff
@@ -210,6 +213,7 @@ func (s *storer) GetListView(ctx context.Context) ([]*experimentation.ListViewIt
 			upper(execution_time),
 			cancellation_time,
 			creation_time,
+		    termination_reason,
```
Comment (snowp) [inline backend/service/chaos/experimentation/experimentstore/storer.go:216]: indentation

File: backend/service/chaos/experimentation/experimentstore/storer.go [hunk 4]
```diff
@@ -270,6 +274,7 @@ func (s *storer) getExperiment(ctx context.Context, runId string) (*Experiment,
...
```
Comment (snowp) [inline backend/service/chaos/experimentation/experimentstore/storer.go:277]: indentation

**Review by snowp — CHANGES_REQUESTED** [review]

File: api/chaos/experimentation/v1/experiment.proto [hunk 5]
```diff
@@ -19,8 +19,6 @@ message Experiment {
     STATUS_COMPLETED = 3;
     // An experiment has been canceled before it started.
     STATUS_CANCELED = 4;
-    // An experiment was running and was stopped before its originally scheduled end time.
-    STATUS_STOPPED = 5;
```
[the remark this answers is not in the document]
Comment (Augustyniak) [reply api/chaos/experimentation/v1/experiment.proto:23]: yes

File: api/chaos/experimentation/v1/experiment.proto [hunk 6]
```diff
@@ -19,8 +19,6 @@ message Experiment {
     STATUS_COMPLETED = 3;
     // An experiment has been canceled before it started.
```
[the remark this answers is not in the document]
Comment (Augustyniak) [reply api/chaos/experimentation/v1/experiment.proto:20]: channged

File: backend/service/chaos/experimentation/experimentstore/experiment.go (hunk 2 above)
Comment (Augustyniak) [reply backend/service/chaos/experimentation/experimentstore/experiment.go:18]: Not a huge fan either but methods that convert to proto are usually called `Proto` or `toProto` so I think it's fine to follow this convention here. I know that our method is not a standard proto conversion but I think it should be fine.

Comment (Augustyniak) [reply backend/service/chaos/experimentation/experimentstore/experiment.go:18]: mostly for testing purposes

File: api/chaos/experimentation/v1/experiment.proto (hunk 5 above)
[the remark this answers is not in the document]
Comment (Augustyniak) [reply api/chaos/experimentation/v1/experiment.proto:23]: It's not used and makes the API more complicated. Now that it's going to be consumed by CI I want to start simple and expand if needed.

**Pull request closed without merging**